mne/decoding/tests/test_time_gen.py

```diff
@@ -70,7 +70,15 @@ def test_generalization_across_time():
     assert_equal("<GAT | fitted, start : -0.200 (s), stop : 0.499 (s), no "
                  "prediction, no score>", '%s' % gat)
     assert_equal(gat.ch_names, epochs.ch_names)
+    # test different predict function:
+    gat.decision_function(epochs)
+    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 1))
+    gat.predict_proba(epochs)
+    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 2))
```

Comment:
A somewhat convoluted way could be
1. probabilities are written to `gat._y_proba_` by `gat.predict_proba`
2a. gat gets a new param `gat.score_probabilities` or something like that, and if `True`, it passes _y_proba to the scorer instead of y_pred
or
2b. gat gets a new methods `gat.score_proba`, which passes _y_proba_ to the scorer instead of _y_pred

That way, you could still use the sklearn metrics if `score_probabilities=False` right?